File: iPhone/VerBank-CSTSv3-IOSClientApi/ClientStation/api/event/API_TradeResultEvent.m
//
//  API_TradeResultEvent.m
//  JEDIv7-CSTSv3-ClientAPI
//
//  Created by baolin on 13/5/17.
//  Copyright (c) 2013年 wangyubo. All rights reserved.
//

#import "API_TradeResultEvent.h"

@implementation API_TradeResultEvent

-(id)init:(IPFather *)ipfahter opfather:(OPFather *)opfather
{
    if (self=[super init]) {
        ipFather=ipfahter;
        opFather=opfather;
        tradeID=[ipfahter getID];
        tradeResult=[opfather isSuccessed];
        errCode=[opfather getErrCode];
    }
    return self;
}

-(NSString *)getTradeID{
    return tradeID;
}

-(Boolean)isTradeResult{
    return tradeResult;
}

-(NSString *)getErrorCode{
    return errCode;
}

-(OPFather *)getOpFather{
    return opFather;
}

-(IPFather *)getIpFather{
    return ipFather;
}
@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/event/data/API_TradeResultEvent.java
package jedi.ex01.client.station.api.event.data;

import jedi.ex01.CSTS3.comm.ipop.IPFather;
import jedi.ex01.CSTS3.comm.ipop.OPFather;
import jedi.ex01.client.station.api.event.API_IDEvent;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class API_TradeResultEvent extends API_IDEvent {
	private String tradeID;
	private boolean tradeResult;
	private String errorCode;

	private OPFather opFather;
	private IPFather ipFather;

	public API_TradeResultEvent(IPFather ipFather, OPFather opFather) {
		super(API_IDEvent_NameInterface.NAME_TRADERETURN);
		this.ipFather = ipFather;
		this.opFather = opFather;
		this.tradeID = ipFather.getID();
		this.tradeResult = opFather.isSucceed();
		errorCode = opFather.getErrCode();
	}

	public String getTradeID() {
		return tradeID;
	}

	public boolean isTradeResult() {
		return tradeResult;
	}

	public String getErrorCode() {
		return errorCode;
	}

	public OPFather getOpFather() {
		return opFather;
	}

	public IPFather getIpFather() {
		return ipFather;
	}
}
